crates/consensus/src/transaction/rlp.rs

```diff
@@ -0,0 +1,206 @@
+
+/// Helper trait for managing RLP encoding of transactions inside 2718
+/// envelopes.
+#[doc(hidden)]
+pub trait RlpEcdsaTx: SignableTransaction<Signature> + Sized {
+    /// The default transaction type for this transaction.
+    const DEFAULT_TX_TYPE: u8;
+
+    /// Calculate the encoded length of the transaction's fields, without a RLP
+    /// header.
+    fn rlp_encoded_fields_length(&self) -> usize;
+
+    /// Encodes only the transaction's fields into the desired buffer, without
+    /// a RLP header.
+    fn rlp_encode_fields(&self, out: &mut dyn alloy_rlp::BufMut);
+
+    /// Create an rlp header for the unsigned transaction.
+    fn rlp_header(&self) -> Header {
+        Header { list: true, payload_length: self.rlp_encoded_fields_length() }
+    }
+
+    /// Get the length of the transaction when RLP encoded.
+    fn rlp_encoded_length(&self) -> usize {
+        self.rlp_header().length_with_payload()
+    }
+
+    /// RLP encodes the transaction.
+    fn rlp_encode(&self, out: &mut dyn BufMut) {
+        self.rlp_header().encode(out);
+        self.rlp_encode_fields(out);
+    }
+
+    /// Create a header for the signed transaction.
+    fn rlp_header_signed(&self, signature: &Signature) -> Header {
+        let payload_length = self.rlp_encoded_fields_length() + signature.rlp_vrs_len();
+        Header { list: true, payload_length }
+    }
+
+    /// Get the length of the transaction when RLP encoded with the given
+    /// signature.
+    fn rlp_encoded_length_with_signature(&self, signature: &Signature) -> usize {
+        self.rlp_header_signed(signature).length_with_payload()
+    }
+
+    /// RLP encodes the transaction with the given signature.
+    fn rlp_encode_signed(&self, signature: &Signature, out: &mut dyn BufMut) {
+        self.rlp_header_signed(signature).encode(out);
+        self.rlp_encode_fields(out);
+        signature.write_rlp_vrs(out);
+    }
+
+    /// Get the length of the transaction when EIP-2718 encoded. This is the
+    /// 1 byte type flag + the length of the RLP encoded transaction.
+    fn eip2718_encoded_length(&self, signature: &Signature) -> usize {
+        self.rlp_encoded_length_with_signature(signature) + 1
+    }
+
+    /// EIP-2718 encode the transaction with the given signature and type flag.
+    fn eip2718_encode_with_type(&self, signature: &Signature, ty: u8, out: &mut dyn BufMut) {
+        out.put_u8(ty);
+        self.rlp_encode_signed(signature, out);
+    }
+
+    /// EIP-2718 encode the transaction with the given signature and the default
+    /// type flag.
+    fn eip2718_encode(&self, signature: &Signature, out: &mut dyn BufMut) {
+        self.eip2718_encode_with_type(signature, Self::DEFAULT_TX_TYPE, out);
+    }
+
+    /// Create a header for the network encoded transaction.
+    fn network_header(&self, signature: &Signature) -> Header {
+        let payload_length = self.eip2718_encoded_length(signature);
+        Header { list: false, payload_length }
+    }
```

Comment:
hmm, but legacy here would be `list: true` right?

Reply:
yep, fixed